target_chains/sui/contracts/sources/price_info.move

```diff
@@ -1,20 +1,104 @@
 module pyth::price_info {
-    use pyth::price_feed::PriceFeed;
+    use sui::object::{Self, UID, ID};
+    use sui::tx_context::{TxContext};
+    use sui::dynamic_object_field::{Self};
+    use sui::table::{Self};
 
+    use pyth::price_feed::{Self, PriceFeed};
+    use pyth::price_identifier::{PriceIdentifier};
+
+    const KEY: vector<u8> = b"price_info";
+
+    friend pyth::pyth;
+
+    /// Sui Object version of PriceInfo.
+    /// Has a key and lives in global store.
+    struct PriceInfoObject has key, store {
```

Comment:
The reasoning was that the code previously made use of `PriceInfo` as a piece of data with the `copy` and `drop` traits. Objects that have the `key` trait cannot have `copy` or `drop`. Wanted to preserve as much existing logic as possible, so made a separate wrapper type `PriceInfoObject` with the `key` ability. We need something with `key` ability, so that it can be a shared object -- accessible by everyone who wants to update the feed it contains.

For example, destroying a [BatchPriceAttestation](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/aptos/contracts/sources/batch_price_attestation.move#L57) produces a vector of `PriceInfos`, which is passed to `update_cache`. After updating the state, those `PriceInfos` are dropped. If `PriceInfo` had key, it could not be dropped.